pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
Yeah, technically the function will return after 168 hours, but the agent will have almost definitely been killed by this point anyway. This way we get killed if we are doing the right thing and exit cleanly if rebooting has an issue, which seems backward to me. The way this chain of functions is currently called, there is no difference between calling regular sleep and calling sleepOrDone, because stop will only be closed after we wait for 168 hours, stop is closed once `Run` returns and `Run` only returns once `process` returns, and process doesn't return until it sleeps for 168 hours.